AcceptanceTest/ATRunner.cs

```diff
@@ -1,64 +1,242 @@
+            }
+
+            processStartInfo.ArgumentList.Add("Runner.rb");
+            processStartInfo.ArgumentList.Add("127.0.0.1");
+            processStartInfo.ArgumentList.Add(port.ToString());
+            processStartInfo.ArgumentList.Add(testDefinitions);
+
+            processStartInfo.RedirectStandardOutput = true;
+            processStartInfo.RedirectStandardError = true;
+
+            Process process = Process.Start(processStartInfo)!;
+
+            using CancellationTokenRegistration reg = cancellationToken.Register(() => process.Kill(entireProcessTree: true));
+
+            async Task ReadStandardOutput()
+            {
+                using FileStream fs = File.Create("TestResult.xml");
+                using StreamWriter sw = new(fs);
+
+                string? line;
+                while ((line = await process.StandardOutput.ReadLineAsync()) != null)
+                {
+                    Console.WriteLine(line);
+                    sw.WriteLine(line);
+                }
+            }
+
+            async Task ReadStandardError()
+            {
+                string? line;
+                while ((line = await process.StandardError.ReadLineAsync()) != null)
+                {
+                    Console.Error.WriteLine(line);
+                }
             }
-            catch (System.Exception e)
+
+            await Task.WhenAll(process.WaitForExitAsync(cancellationToken), ReadStandardOutput(), ReadStandardError());
+
+            return process.ExitCode;
+        }
+
+        private static void PrintResults(string testDefinitions, bool printTable)
+        {
+            Xml.AcceptanceTest at;
+            XmlSerializer xmlSerializer = new(typeof(Xml.AcceptanceTest));
+            using (FileStream fs = File.OpenRead("TestResult.xml"))
             {
-                debugLog.OnEvent(e.ToString());
+                at = (Xml.AcceptanceTest)xmlSerializer.Deserialize(fs)!;
             }
 
-            finally
+            if (printTable)
             {
-                if(acceptor != null)
-                    acceptor.Stop();
+                int maxWidth = at.Tests.Max(t => Path.GetFileName(t.Name).Length);
+
+                Console.WriteLine($"{"Name".PadRight(maxWidth)} Result  Message");
+                Console.WriteLine($"{"----".PadRight(maxWidth)} ------  -------");
+
+                foreach (Test t in at.Tests)
+                {
+                    Console.WriteLine($"{Path.GetFileName(t.Name).PadRight(maxWidth)} {t.Result} {t.Message?.Text.Trim()}");
+                }
+            }
+
+            string? version = Path.GetFileName(Path.GetDirectoryName(testDefinitions));
+
+            if (version is not null &&
+                Regex.Match(version, @"(?in)^fix(?<Major>\d)(?<Minor>\d)(?<ServicePack>(?<Prefix>SP)(?<Number>\d+))?$") is Match { Success: true } m)
+            {
+                // Format the version e.g. "fix50sp2" => "FIX 5.0 SP 2"
```

Comment:
You can delete this block if you want.  I don't see any tangible benefit to pretty-formatting a string that isn't even ugly to begin with, and that Regex isn't fun to read.